packages/strapi-plugin-graphql/services/Resolvers.js

```diff
@@ -371,56 +371,16 @@ module.exports = {
             if (association.type === 'model') {
               params[ref.primaryKey] = _.get(obj, [association.alias, ref.primaryKey], obj[association.alias]);
             } else {
-              // Apply optional arguments to make more precise nested request.
-              const convertedParams = strapi.utils.models.convertParams(
-                name,
-                Query.convertToParams(Query.amountLimiting(options)),
-              );
-
-              const where = strapi.utils.models.convertParams(
-                name,
-                options.where || {},
-              );
-
-              // Limit, order, etc.
-              Object.assign(queryOpts, convertedParams, { where: where.where });
-
-              // Skip.
-              queryOpts.skip = convertedParams.start;
-
-              switch (association.nature) {
-                case "manyToMany": {
-                  const arrayOfIds = (obj[association.alias] || []).map(
-                    related => {
-                      return related[ref.primaryKey] || related;
-                    }
-                  );
-                  
-                  Object.assign(queryOpts, {
-                    ...queryOpts,
-                    query: {
-                      [ref.primaryKey]: arrayOfIds
-                    }
-                  });
-
-                  break;
-                }
-                default:
-                  Object.assign(queryOpts, {
-                    ...queryOpts,
-                    query: {
-                      [association.via]: obj[ref.primaryKey]
-                    }
-                  });
-              }
-            }
+              const queryParams = Query.amountLimiting(options);
```

Comment:
Hm... again this part was tricky and the switch hasn't been coded for no reason. It seems almost impossible to me that this part can be so simplified. I would love to but I need to check all those tricky cases again before allowing this PR to be merged.

Reply:
Ya go ahead, I already did some QA with some tricky cases and it seems to be working as excepted, I probably missed something but ya I encourage you to push the QA before merging this PR because it changes a lot of things